examples/all-clusters-app/all-clusters-common/src/laundry-washer-controls-delegate-impl.cpp

```diff
@@ -54,3 +55,44 @@ CHIP_ERROR LaundryWasherControlDelegate::GetSupportedRinseAtIndex(size_t index,
     supportedRinse = LaundryWasherControlDelegate::supportRinsesOptions[index];
     return CHIP_NO_ERROR;
 }
+
+Status LaundryWasherControlDelegate::PreAttributeCheck(AttributeId attributeId, uint16_t size, uint8_t * value)
+{
+    switch (attributeId)
+    {
+    case Attributes::SpinSpeedCurrent::Id: {
+        if (*value >= ArraySize(spinSpeedsNameOptions))
+        {
+            return Status::ConstraintError;
+        }
+        return Status::Success;
+    }
+    case Attributes::NumberOfRinses::Id: {
+        if (size != sizeof(NumberOfRinsesEnum))
+        {
+            return Status::InvalidDataType;
+        }
+        for (uint16_t index = 0; index < ArraySize(supportRinsesOptions); index++)
+        {
+            if (static_cast<NumberOfRinsesEnum>(*value) == LaundryWasherControlDelegate::supportRinsesOptions[index])
```

Comment:
OK, so now this is assuming that `sizeof(NumberOfRinsesEnum) == 1`, right?  That should be a static_assert.

Again, why is this check not just done in the cluster?

Reply:
Will move to the cluster